### ISSUE TITLE
Wishing to make it run on MacOS

### ISSUE THREAD

**Issue by hasantahir** [ahrm/sioyek#231]
Hi, 

I have been a long time admirer of the app but haven't managed to make it work ever since I saw it on the mac apps subreddit. I am trying to build the app from the source on my mac. I have all the dependencies and prerequisites installed. I get the following error during the build process:

```
ld: library not found for -lmupdf-third
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [sioyek.app/Contents/MacOS/sioyek] Error 1
ERROR: Could not find bundle binary for "build/sioyek.app"
ERROR: "error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file:  (No such file or directory)\n"
ERROR: "error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file:  (No such file or directory)\n"
ERROR: "error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file:  (No such file or directory)\n"
WARNING:
WARNING: Could not find any external Qt frameworks to deploy in "build/sioyek.app"
WARNING: Perhaps macdeployqt was already used on "build/sioyek.app" ?
WARNING: If so, you will need to rebuild "build/sioyek.app" before trying again.
ERROR: Could not find bundle binary for "build/sioyek.app"
ERROR: "error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/strip: can't open file:  (No such file or directory)\n"
ERROR: ""
  adding: build/sioyek.dmg (deflated 5%)
  ```
  
  I have the `mupdf` version 1.19.1 installed through Homebrew.
  
  macOS version is 12.3.1 with XCode 13.3.1 installed.
  
  Also wanted to mention that the prebuilt binary version supplied on Github hangs the system once I open the app. 
  
  Edit: Latest prebuilt version throws the following error:
  
  
<img width="532" alt="Screenshot 2022-05-16 at 12 08 33" src="https://user-images.githubusercontent.com/11236148/168580086-c4d12681-7a68-4142-8606-c16881fadccb.png">

  
  Thanks and appreciate any help.

**Comment by ahrm** [comment]
That is strange. You can see the exact script we run to build the mac release here:
https://github.com/ahrm/sioyek/blob/db83becfc2232eece486c0cabb7fbbe604051996/.github/workflows/build_and_release.yml#L246 . Are you following the same steps?

Also as a shot in the dark you could remove `-lmupdf-third` from this line:
https://github.com/ahrm/sioyek/blob/db83becfc2232eece486c0cabb7fbbe604051996/pdf_viewer_build_config.pro#L105
I probably would cause other problems though.

By the way does the prebuilt binary not work despite the error? For example what happens when you try to open a new file by pressing `o`?

**Comment by ahrm** [comment]
Maybe remove `-lmupdf-threads` as well as see what happens 😬.

**Comment by hasantahir** [comment]
Throwing even more strange errors 😄 . On the pre-built version, nothing happens when I press external search key combinations `ss` or `sl`. Does it need any additional permissions? Thanks.

**Comment by ahrm** [comment]
I am not sure. I don't have a mac computer so I can't test it right now. Maybe someone else can help?

**Comment by ahrm** [comment]
The problem with external search should be fixed in f9a4cbf50a4c4bbd356eea38bffba2df0eb0fb02. Here is an experimental build with latest fixes applied:
https://github.com/hexomancer/sioyek/releases/tag/v0.31.330

**Comment by hasantahir** [comment]
Perfect, the external (Google and Google scholar) search is working. Need to change the URL for `libgen` as it is blocked by ISP.
